py-polars/polars/dataframe/frame.py

```diff
@@ -4753,8 +4753,13 @@ def get_column_index(self, name: str) -> int:
         ... )
         >>> df.get_column_index("ham")
         2
+        >>> df.get_column_index("sandwich")  # doctest: +SKIP
+        ColumnNotFoundError: 'sandwich' not found in DataFrame
         """
-        return self._df.get_column_index(name)
+        if (idx := self._df.get_column_index(name)) is None:
+            msg = f"{name!r} not found in DataFrame"
+            raise ColumnNotFoundError(msg)
+        return idx
```

Comment:
Just wondering: why this is implemented in Python and not on the Rust side? 

I help developing `r-polars` and putting this in Rust would ensure that the Python and R implementations have the same messages, even if the message is updated later.

Reply:
On the Rust side you have `try_get_column_index` for this purpose 👌

Reply:
Yes, but that it's the `polars-core` crate, while we mimic the `py-polars` crate. We also use 

```rs
    pub fn get_column_index(&self, name: &str) -> Option<usize> {
        self.df.get_column_index(name)
    }
```
and we call this from R. I guess my point is more, what isn't the body of the function above `self.df.try_get_column_index(name)` instead? There are already many functions for which the error message is implemented in Rust.

I get that it's not your job to worry about the R implementation, but any Python code that could be moved to the Rust internals makes it harder to have consistent bindings ;)

Reply:
Yes, this method should just map to `try_get_column_index` in the PyO3 bindings.

Reply:
> I get that it's not your job to worry about the R implementation, but adding stuff on the Python side rather than putting it in the Rust internals (if possible) makes it harder to have consistent bindings ;)

Makes perfect sense; the more consistent, the better 😉